src/util.cpp

```
@@ -667,76 +667,118 @@ void CLanguageComboBox::OnLanguageActivated ( int iLanguageIdx )
+    // parse input address with rx1 and rx2 in turn, capturing address/host and port
+    if ( rx1.indexIn ( strAddress ) == 0 )
     {
-        // IP6 address
-        bIsIP6 = true;
-        strSep = "]:";
+        // literal address within []
+        strAddress   = rx1.cap ( 1 );
+        strPort      = rx1.cap ( 2 );
+        bLiteralAddr = true; // don't allow hostname within []
+    }
+    else if ( rx2.indexIn ( strAddress ) == 0 )
+    {
+        // hostname or IPv4 address
+        strAddress = rx2.cap ( 1 );
+        strPort    = rx2.cap ( 2 );
+    }
+    else
+    {
+        // invalid format
+        // qInfo() << qUtf8Printable ( QString ( "Invalid address format" ) );
+        return false;
     }
-
-    QString strPort = strAddress.section ( strSep, 1, 1 );
 
     if ( !strPort.isEmpty() )
     {
-        // a colon is present in the address string, try to extract port number
+        // a port number was given: extract port number
         iNetPort = strPort.toInt();
 
-        // extract address port before separator (should be actual internet address)
-        strAddress = strAddress.section ( strSep, 0, 0 );
-
-        if ( bIsIP6 )
+        if ( iNetPort >= 65536 )
         {
-            // remove "[" at the beginning
-            strAddress.remove ( 0, 1 );
+            // invalid port number
+            // qInfo() << qUtf8Printable ( QString ( "Invalid port number specified" ) );
+            return false;
         }
     }
 
     // first try if this is an IP number an can directly applied to QHostAddress
-    if ( !InetAddr.setAddress ( strAddress ) )
+    if ( InetAddr.setAddress ( strAddress ) )
     {
-        // it was no valid IP address, try to get host by name, assuming
+        if ( !bEnableIPv6 && InetAddr.protocol() == QAbstractSocket::IPv6Protocol )
+        {
+            // do not allow IPv6 addresses if not enabled
+            // qInfo() << qUtf8Printable ( QString ( "IPv6 addresses disabled" ) );
```

Do we give an error message to the user somewhere else?

---

At the moment, with all versions, if an invalid hostname or IP address is entered into the Connect window, the connection fails silently. It would be good to fix that more generally, but I think that should be separate from this PR. Maybe raise it as a specific issue?

---

Done #1945